var/spack/repos/builtin/packages/aomp/package.py

```diff
@@ -4,104 +4,121 @@
+    "3ff18a8bd31d5b55232327e574dfa3556cf26787e105d0ba99411c5687325a8d"
+]
+
+flang = [
+    "cc27f8bfb49257b7a4f0b03f4ba5e06a28dcb6c337065c4201b6075dd2d5bc48",
+    "1fe07a0da20eb66a2a2aa8d354bf95c6f216ec38cc4a051e98041e0d13c34b36",
+    "54cc6a9706dba6d7808258632ed40fa6493838edb309709d3b25e0f9b02507f8",
+    "43d57bcc87fab092ac242e36da62588a87b6fa91f9e81fdb330159497afdecb3",
+    "81674bf3c9d8fd9b16fb3e5c66a870537c25ff8302fc1b162ab9e95944167163"
+]
+
+extras = [
+    "5dbf27f58b8114318208b97ba99a90483b78eebbcad4117cac6881441977e855",
+    "adaf7670b2497ff3ac09636e0dd30f666a5a5b742ecdcb8551d722102dcfbd85",
+    "4460a4f4b03022947f536221483e85dcd9b07064a54516ec103a1939c3f587b5",
+    "014fca1fba54997c6db0e84822df274fb6807698b6856da4f737f38f10ab0e5d",
+    "ee146cff4b9ee7aae90d7bb1d6b4957839232be0e7dab1865e0ae39832f8f795"
+]
+
+# Used only for 3.5.0
+hip = [
+    "86eb7749ff6f6c5f6851cd6c528504d42f9286967324a50dd0dd54a6a74cacc7"
+]
+
+vdi = [
+    "b21866c7c23dc536356db139b88b6beb3c97f58658836974a7fc167feb31ad7f"
+]
+
+opencl = [
+    "8963fcd5a167583b3db8b94363778d4df4593bfce8141e1d3c32a59fb64a0cf6"
+]
+
+versions = ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0']
+versions_dict = dict()
+hashes = [aomp, devlib, llvm, flang, extras]
+hashes_35 = [aomp, devlib, llvm, flang, extras, hip, vdi, opencl]
+components = ['aomp', 'devlib', 'llvm', 'flang', 'extras']
+components_35 = [
+    'aomp', 'devlib', 'llvm', 'flang', 'extras', 'hip', 'vdi', 'opencl'
+]
+
+for outer_index, item in enumerate(versions):
+    if item == '3.5.0':
+        use_components = components_35
+        use_hashes = hashes_35
+    else:
+        use_components = components
+        use_hashes = hashes
+    for inner_index, component in enumerate(use_hashes):
+        versions_dict.setdefault(item, {})[use_components[inner_index]] = \
+            use_hashes[inner_index][outer_index]
 
 
 class Aomp(Package):
     """llvm openmp compiler from AMD."""
 
     homepage = tools_url + "/aomp"
-    url = tools_url + "/aomp/archive/rocm-3.9.0.tar.gz"
+    url = tools_url + "/aomp/archive/rocm-3.10.0.tar.gz"
 
     maintainers = ['srekolam', 'arjun-raj-kuppala', 'estewart08']
-    version('3.9.0', sha256=aomp38['aomp'])
-    version('3.8.0', sha256=aomp38['aomp'])
-    version('3.7.0', sha256=aomp37['aomp'])
-    version('3.5.0', sha256=aomp35['aomp'])
+    version('3.10.0', sha256=versions_dict['3.10.0']['aomp'])
+    version('3.9.0', sha256=versions_dict['3.9.0']['aomp'])
+    version('3.8.0', sha256=versions_dict['3.8.0']['aomp'])
+    version('3.7.0', sha256=versions_dict['3.7.0']['aomp'])
+    version('3.5.0', sha256=versions_dict['3.5.0']['aomp'])
+
+    # Cmake above 3.18 would fail the build on 3.5.0
+    depends_on('cmake@3:', type='build')
+    depends_on('cmake@3:3.17.4', when='@3.5.0', type='build')
+
+    # Python 2 is needed for 3.5.0 and 3.8.0, limit py-setuptools
+    # to avoid spec error
+    depends_on('python@2.7.8:2.7.18', when='@3.5.0:3.8.0', type='build')
```

Comment:
Python 2.7.7 doesn't work?
```suggestion
    depends_on('python@2.7:2.8', when='@3.5.0:3.8.0', type='build')
```

Reply:
done